flask_resources/errors.py

```diff
@@ -33,6 +33,43 @@ def wrapper(self, *args, **kwargs):
     return wrapper
 
 
+class ErrorMapper:
+    """Class to map business logic errors to REST exceptions."""
+
+    exception_map = {}
+
+    def __init__(self, func):
+        """Constructor."""
+        update_wrapper(self, func)
+        self.func = func
+
+    def __call__(self, *args, **kwargs):
+        """Raise the mapped exception based on the `exception_map`."""
+        try:
+            return self.func(*args, **kwargs)
+        except tuple(self.exception_map.keys()) as exc:
+            exc_or_func = self.exception_map[exc.__class__]
+            if callable(exc_or_func) and not isinstance(exc_or_func, HTTPException):
+                exc_or_func = exc_or_func(exc)
+            raise exc_or_func from exc
```

Comment:
This is very fancy... why not a function that does this? (curious)

Reply:
Sorry, I didn't understand which part of the code do you mean... Do you mean the `raise` part?

Reply:
I mean the usage of a class with `__call__` + `update_wrapper` . I deem those more advanced tools. Curious why more advanced tools were used.

Reply:
@fenekku because I wanted to use an interface i.e a class to represent the error mapping. My idea is that people will just inherit/instantiate the `ErrorMapper` class with a custom `exception_map`. Then, `__call__` method was used so this class can act as a decorator class.

Reply:
But the `callable` check here was added because in your exception_map you could do something like
```python
exception_map = {
  NotFound: lambda exc: MyException(description=exc.description)
}
```
and consume information of the original exception when you map to your REST exception!

Reply:
Thanks for the info! I am wondering about how to make the configuration more uniform for the devs defining it, but it's a nebulous thought that can be addressed later.